src/macros.rs

```diff
@@ -23,12 +23,7 @@ macro_rules! log {
             __module_path: module_path!(),
         };
         let lvl = $lvl;
-        if !cfg!(log_level = "off") &&
-                (lvl <= $crate::LogLevel::Error || !cfg!(log_level = "error")) &&
-                (lvl <= $crate::LogLevel::Warn || !cfg!(log_level = "warn")) &&
-                (lvl <= $crate::LogLevel::Debug || !cfg!(log_level = "debug")) &&
-                (lvl <= $crate::LogLevel::Info || !cfg!(log_level = "info")) &&
-                lvl <= $crate::max_log_level() {
+        if lvl <= $crate::__static_max_level() && lvl <= $crate::max_log_level() {
```

Comment:
Should we worry about breaking back-compat for people that are actually using the log_level cfg? I don't know if any of those people exist, but might be worth leaving this around to support them? Thoughts @alexcrichton?